samcli/commands/init/__init__.py

```diff
@@ -64,21 +65,31 @@ def cli(ctx, location, runtime, output_dir, name, no_input):
         \b
         $ sam init --location /path/to/template/folder
 
+        \b
+        Initializes a new SAM project using custom template with additional known context
+        \b
+        # Add multiple -e for multiple context to be added (-e some=Value -e other=Value)
+        $ sam init --location gh:aws-samples/cookiecutter-aws-sam-python -e project_short_description=Oi
+
     """
     # All logic must be implemented in the `do_cli` method. This helps ease unit tests
     do_cli(ctx, location, runtime, output_dir,
-           name, no_input)  # pragma: no cover
+           name, no_input, extra_context)  # pragma: no cover
 
 
-def do_cli(ctx, location, runtime, output_dir, name, no_input):
+def do_cli(ctx, location, runtime, output_dir, name, no_input, extra_context):
     """
     Implementation of the ``cli`` method, just separated out for unit testing purposes
     """
     LOG.debug("Init command")
     click.secho("[+] Initializing project structure...", fg="green")
 
+    # Break up given extra context into dict(key:value); else use a sane default empty dict
+    options = (opts.split('=') for opts in extra_context)
```

Comment:
question: will extra_context always be of form a=b and thats it?

will we ever do a=b,c=d. I think we need something similair to that in !Ref or env var subsitution. I fail to recall which.

Reply:
Great point. I took inspiration from other CLIs like docker-compose as that's easily readable when you have multiple params. 

I looked into the [SAM CLI code and the closest we do is for env-vars option](https://github.com/awslabs/aws-sam-cli/blob/f88fb592e072314a4902e1f00489329a58c79519/samcli/commands/local/lib/local_lambda.py#L119) and that expects a file in JSON format. ``sam deploy`` shells out to AWS CLI which in turn follows `a=b,c=d`. 

I'm happy either way:

A) Implement `a=b,c=d` format like AWS CLI does
B) Reuse what ``--env-vars`` does for JSON file like ``--env-vars`` and disallow input via arguments

Let me know your thoughts _and_ thanks for taking the time to review this!